Wrappers/Python/cil/optimisation/algorithms/ISTA.py

```diff
@@ -60,8 +71,25 @@ class ISTA(Algorithm):
     >>> ig = Aop.domain
     >>> ista = ISTA(initial = ig.allocate(), f = f, g = g, max_iteration=10)     
     >>> ista.run()
+
+
+    See also
+    --------
+
+    :class:`.FISTA`
+
+    :class:`.GD`
+
       
     """
+
+    def check_stepsize(self):
+        if isinstance(self.step_size, Number) and self.step_size<=0:
+            raise ValueError("Positive step size is required. Got {}".format(self.step_size))
+
+    def check_convergence(self):
```

Comment:
I would rather call this `check_convergence_criterion`, as with `check_convergence` it looks like the function checks that the algorithm is converged.

Reply:
I would call it `is_convergent` because it returns a boolean telling us if the algorithm config fits with the convergence criterion. https://github.com/TomographicImaging/CIL/pull/1143#discussion_r830020326

Reply:
In my opinion, `is_convergent` refers to the algorithm, but here we need to check if the proved criterion for this algorithm is true or not. I think we need the word `criterion`.